### PR TITLE
Set compiler config to be owned (following wasm-c-api)

### PR DESCRIPTION
**Pull request by syrusakbary**
# Description

Set compiler config to be owned (following wasm-c-api)

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by syrusakbary** [comment]
bors try

File: lib/compiler-llvm/src/config.rs [hunk 1]
```diff
@@ -207,8 +207,8 @@ impl CompilerConfig for LLVM {
     }
 
     /// Transform it into the compiler.
-    fn compiler(&self) -> Box<dyn Compiler + Send> {
-        Box::new(LLVMCompiler::new(&self))
+    fn compiler(self: Box<Self>) -> Box<dyn Compiler + Send> {
```
Comment (MarkMcCaskey) [inline lib/compiler-llvm/src/config.rs:210]: Consider using `From`/`Into` traits here; not sure how this is used but that could be a bit cleaner

**Review by MarkMcCaskey — APPROVED** [review]
Looks good to me

File: lib/compiler-llvm/src/config.rs (hunk 1 above)
Comment (syrusakbary) [reply lib/compiler-llvm/src/config.rs:210]: I tried, couldn't make it work with `From`. We can do it in a PR apart if necessary

**Comment by syrusakbary** [comment]
Merging directly to alleviate CI. Tests are passing locally

**Pull request merged**